[docs] Improve license installation page (#16403)
mui/mui-x: docs/data/introduction/licensing/licensing.md

### Where to install the key

> The license key needs to be set during run-time, not build-time, before any MUI X component is rendered in your app using the `setLicenseKey()` function.

Reviewer: I think this an overcorrection. The previous message is more likely to be what people fail with.

Author: Yes, likely. That's why I want the input from @mapache-salvaje for a better wording (I am pretty bad at this! 🤣)

Reviewer 2: How about something like this? 

(By the way, we need to standardize our usage of "runtime" vs "run-time" - I'll add this to the company style guide, but it should be "runtime" when used as a noun as it is here, and "run-time" when used an adjective, e.g. "run-time performance.")

```suggestion
You must call the `setLicenseKey()` function during runtime—not build time—before any React component is rendered in your app.
```

Reviewer: Hum, what I had in mind was this order of likelihood for the installation to fail:

1. Set too late
2. Never set
3. Set somewhere else, e.g. built time.

```suggestion
You must call the `setLicenseKey()` function before React renders the first component in your app. 

Because the license is verified when the components mount, this function must be called in the browser (e.g. calling it inside next.config.js wouldn't work).
```

Reviewer 2: That sounds good to me, I would just make a few small style changes:

```suggestion
You must call the `setLicenseKey()` function before React renders the first component in your app.
Because the license is verified when the components mount, this function must be called in the browser (which means, for example, that calling it inside `next.config.js` won't work).```

### Next.js Pages Router

When using Next.js pages, a great place to call `setLicenseKey` is in [`_app.tsx`](https://nextjs.org/docs/pages/building-your-application/routing/custom-app).

```tsx title="pages/_app.tsx"

> import type { ReactElement, ReactNode } from 'react';

Reviewer: ```suggestion
import * as React from 'react';
```

import type { NextPage } from 'next';
import type { AppProps } from 'next/app';
import { LicenseInfo } from '@mui/x-license';

LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');

export type NextPageWithLayout<P = {}, IP = P> = NextPage<P, IP> & {

>   getLayout?: (page: ReactElement) => ReactNode;

Reviewer: ```suggestion
  getLayout?: (page: React.ReactElement) => React.ReactNode;
```

};

type AppPropsWithLayout = AppProps & {
  Component: NextPageWithLayout;
};

> export default function MyApp({ Component, pageProps }: AppPropsWithLayout) {

Reviewer: I have been constantly normalizing towards having all our code to not destrructre the props in the argument side, to make it super clear that this is a component.

```suggestion
export default function MyApp(props: AppPropsWithLayout) {
  const { Component, pageProps } = props;
```
